public/app/features/stars/StarToolbarButton.tsx

```diff
@@ -63,18 +64,29 @@ export function StarToolbarButtonApiServer({ group, kind, id }: Pick<Props, 'gro
 }
 
 function StarToolbarButtonLegacy({ dashboard }: { dashboard: Props['dashboard'] }) {
-  const { meta } = dashboard.useState();
-  const isStarred = Boolean(meta.isStarred);
+  const { meta, uid: uidFromState } = dashboard.useState();
+  // uidFromState is used for legacy dashboards (kubernetesDashboards toggle is off)
+  const uid = meta.uid || meta.k8s?.name || uidFromState;
   const tooltips = getStarTooltips();
 
+  const { value, retry } = useAsyncRetry(async () => {
```

Comment:
nit: on naming, can `value` be more descriptive? Like `usersStars` or `starredDashboardIds`?